app/scripts/lib/derived-keys.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Given the technical detail, I think more inline docs on the params of each method and their expected types would be useful. It could be a follow on PR.